R/conformal_infer.R

```diff
@@ -0,0 +1,455 @@
+#' Prediction intervals via conformal inference
+#'
+#' Nonparametric prediction intervals can be computed for fitted workflow
+#' objects using the conformal inference method described by Lei _at al_ (2018).
+#'
+#' @param object A fitted [workflows::workflow()] object.
+#' @param control A control object from [control_conformal_infer()] with the
+#' numeric minutiae.
+#' @param ... Not currently used.
+#' @param train_data A data frame with the _original predictor data_ used to
+#' create the fitted workflow (predictors and outcomes). If the workflow does
+#' not contain these values, pass them here. If the workflow used a recipe, this
+#' should be the data that were inputs to the recipe (and not the product of a
+#' recipe). It should be a named argument.
+#' @return An object of class `"int_conformal_infer"` containing the information
+#' to create intervals (which includes the training set data). The `predict()`
+#' method is used to produce the intervals.
+#' @details
+#' This function implements what is usually called "full conformal inference"
+#' (see Algorithm 1 in Lei _et al_ (2018)) since it uses the entire training
+#' set to compute the intervals.
+#'
+#' This function prepares the objects for the computations. The [predict()]
+#' method computes the intervals for new data.
+#'
+#' For a given prediction, different "trial" values of the outcome are evaluated
+#' but augmenting the training set with the sample being predicted (with the
+#' trial value as the outcome). From this model, the residual associated with
+#' the trial value is compared to a quantile of the distribution of the other
+#' residuals. Usually the absolute values of the residuals are used. Once the
```

Comment:
Was originally unclear to me whether the "other residuals" were the residuals associated with the original training data for a given observation or the residuals associated with the other trial values across models.

Reply:
The residuals from the (real) training set, meaning the non-trial values